lib/phlex/translation.rb

```diff
@@ -10,8 +10,15 @@ module ClassMethods
 			attr_writer :translation_path
 
 			def translation_path
-				@translation_path ||= name&.split("::")&.join(".")&.downcase.to_s
-			end
+        @translation_path ||= name
+                                &.split("::")
+                                &.join(".")
+                                &.gsub(/([A-Z]+)([A-Z][a-z])/,'\1_\2')
+                                &.gsub(/([a-z\d])([A-Z])/,'\1_\2')
+                                &.tr("-", "_")
+                                &.downcase.to_s
```

Comment:
I think this will be faster and do the same thing. It only allocates one String rather than 1 Array and 4 Strings.

```suggestion
        @translation_path ||= name&.dup.tap do |n|
          n.gsub!("::", ".")
          n.gsub!(/([a-z])([A-Z])/, '\1_\2')
          n.downcase!
        end
```

Reply:
Oh, that's very true.